src/nodes.coffee

```diff
@@ -2029,14 +2136,18 @@ exports.Code = class Code extends Base
     @eachParamName (name, node, param) ->
       node.error "multiple parameters named '#{name}'" if name in paramNames
       paramNames.push name
-
       if node.this
         name   = node.properties[0].name.value
         name   = "_#{name}" if name in JS_FORBIDDEN
         target = new IdentifierLiteral o.scope.freeVariable name
         param.renameParam node, target
         thisAssignments.push new Assign node, target
-
+      # In case the parameter is object and contains the splat, e.g. fn({a, b, r...})
+      # Can be removed once ES proposal hits Stage 4.  
+      if param.name instanceof Obj and param.name.hasSplat()
```

Comment:
Without this `foo({a, b, ...r})` compiles to `foo(Object.assign({}, {a, b}, r))`